### ISSUE TITLE
Corrupted GIF at the output

### ISSUE THREAD

**Issue by photopea** [deanm/omggif#43]
This image has 190 different colors. 

![head](https://user-images.githubusercontent.com/15955558/139122566-f23e93a4-1993-43da-b440-5527afdcea9c.png)

When I encode it into GIF using GifWriter ("as it is"), the resulting GIF is corrupted. The LZW stream is damaged.

![head](https://user-images.githubusercontent.com/15955558/139122529-7123d576-4b14-46ba-ad43-2d82283bbe9c.gif)

Could you fix it?

**Comment by deanm** [comment]
Do you have any more details about how / why / where it is corrupt? I haven't seen any similar bugs in LZW encoding, is it possible it's somehow an issue in how it's being used? Or could you narrow down the issue a bit more than "it doesn't work please fix it"? Thanks!

**Comment by photopea** [comment]
You can reproduce the issue this way:

    var buf = new Uint8Array(1000000);
    var gw = new GifWriter(buf, 145, 91, {palette:plte, loop:0});
    gw.addFrame(0,0,145,91, new Uint8Array(img), {transparent:0, disposal:2, delay:2});
    var gif = buf.slice(0,gw.end());

The values of "plte" and "img" can be found here:  https://pastebin.com/ede3mV5d

The resulting GIF has an error in the last line of pixels. E.g. the bottom right pixel cannot be decoded by any GIF viewer.

**Comment by deanm** [comment]
I'm not sure what you're doing here.

1) Your example code doesn't produce the output you attached. The output you attached is a gif file 15195 bytes long, this seems to be wrongly truncated, it does not end in the 0x3b trailer byte that would be written by end().

2) The code you sent produces a valid gif file of 15472 bytes.

3) The image produced by your code does not look visually correct however. I don't have the time to debug this for you, but you've already made so many errors along the way that I am pretty sure omggif is not at fault here. If you look at the palette you are supplying, it is not 24-bit RBG, it looks like you are using 32-bit with an alpha channel. This may work, but you should be aware that you must be careful as for bitwise operations as defined by JavaScript (nothing to do with omggif) these will be treated as *signed* 32-bit numbers.

4) You are passing `transparent: 0`, meaning you want the 0 colour index to mean transparent, however you don't seem to actually want transparency here, so you are getting incorrect rendering. As you can read in the code, you should pass either null or undefined (or just don't pass anything, which will be undefined) if you don't actually want transparency.

In summary, the gif you attached does not match the output when I run your the code, when I run the code you attached I get a valid (but most likely visually not what you want) gif. From omggif's perspective I don't see anything wrong here, just that you are not using it correctly. I do not see any LZW encoding errors here, I simply see a truncated file.

**Comment by photopea** [comment]
Man, I am **soooo sorry** :( You helped me find a bug in my code.

Turns out, in the "new GifWriter(buf ...", I was giving your library a "buf" of the length 15195 Bytes. As this image contains 13195 pixels, I thought that IMAGE_AREA + 2000 Bytes would be enough for the output image. I did not expect the GIF to be so much larger than the uncompressed image. And your library did not tell me that the output array is too short.

Thank you again for this great library!
